### PR TITLE
Standardize hover and active states across components

### PR DESCRIPTION
**Pull request by lyubomir-popov**
## Done

[List of work items including drive-bys]

Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/3665

## QA

- Inspect side navigation, chips, buttons (all flavours), accordion
- Verify states are consistent

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical-web-and-design/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical-web-and-design/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [component status page](https://github.com/canonical-web-and-design/vanilla-framework/blob/master/templates/docs/component-status.md).
- [x] Documentation side navigation should be updated with the relevant labels.

## Screenshots

[if relevant, include a screenshot or screen capture]

### CODE REVIEW

File: scss/_patterns_subnav.scss [hunk 1]
```diff
@@ -178,11 +190,12 @@
...
   @include vf-subnav-theme(
-    $color-subnav-icon: #999,
     $color-subnav-background: $colors--dark-theme--background-alt,
+    $color-subnav-background-hover: $colors--dark-theme--background-hover,
+    $color-subnav-background-active: $colors--dark-theme--background-active,
     $color-subnav-text: $colors--dark-theme--text-default,
-    $color-subnav-text-hover: $colors--dark-theme--background-default,
-    $color-subnav-background-hover: $colors--dark-theme--text-default,
+    $color-subnav-text-hover: $colors--dark-theme--text-default,
+    $color-subnav-icon: #999,
```
Comment (bartaz) [inline scss/_patterns_subnav.scss:198]: Why hardcoding the color without any variable?

**Comment by bartaz** [comment]
In sub-navigation it seems that the new active state color works inside the dropdown, but not in the main nav elements. It this intended?

subnav-active

File: scss/_patterns_subnav.scss (hunk 1 above)
Comment (lyubomir-popov) [reply scss/_patterns_subnav.scss:199]: fixed

File: scss/_patterns_navigation.scss [hunk 2]
```diff
@@ -361,23 +371,24 @@ $lightness-threshold: 70;
...
   }
 
   %navigation-link-selected {
-    @include vf-highlight-bar($color-navigation-highlight, left, true, $sph-inner);
+    @extend %navigation-link-hover;
+
+    @include vf-highlight-bar($color-navigation-highlight, left, true, 0);
 
     @media (min-width: $breakpoint-navigation-threshold) {
-      @include vf-highlight-bar($color-navigation-highlight, bottom, false, $sph-inner);
```
Comment (bartaz) [inline scss/_patterns_navigation.scss:372]: It seems that we removed the padding from highlight bar everywhere - can we remove it from mixin as well? It unnecessarily complicates it.

File: scss/_patterns_tabs.scss [hunk 3]
```diff
@@ -71,18 +71,17 @@
...
       &:active,
-      &:hover,
       &[aria-selected='true'] {
-        @include vf-highlight-bar($color-tabs-active-bar, bottom, false, $sph-inner);
+        @include vf-highlight-bar($color-tabs-active-bar, bottom, false, 0);
 
         // Display the highlight when focussing (in combination with the parent
         // states) in modern browsers that support focus-visible.
         &:focus:not(:focus-visible) {
-          @include vf-highlight-bar($color-tabs-active-bar, bottom, false, $sph-inner);
```
Comment (bartaz) [inline scss/_patterns_tabs.scss:85]: As mentioned in other comment, these 0s at the end param can probably be removed, and I guess this padding param is not used anywhere anymore, so could be removed from code completely.

**Comment by lyubomir-popov** [comment]
@bartaz thanks, well spotted. I've removed the unnecessary arg

File: scss/_settings_placeholders.scss [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 // Global placeholder settings
 
-$bar-thickness: 0.1875rem !default; // 3px at 16px fontsize, expressed in rems so it scales with text if the root font-size changes at a breakpoint
+$bar-thickness: 0.1875rem !default; // 4px at 16px fontsize, expressed in rems so it scales with text if the root font-size changes at a breakpoint
```
[the remark this answers is not in the document]
Comment (lyubomir-popov) [reply scss/_settings_placeholders.scss:6]: fixed

**Review by bartaz — APPROVED** [review]
LGTM, thanks

**Pull request merged**